frontend/packages/dev-console/src/components/gitops/PodRingWrapper.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { PodModel } from '@console/internal/models';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind, modelFor } from '@console/internal/module/k8s';
+import { PodRing, ExtPodKind } from '@console/shared';
+import { LoadingInline } from '@console/internal/components/utils';
+import * as _ from 'lodash';
+
+interface PodRingWrapperProps {
+  workload: { [key: string]: any };
+}
+
+const PodRingWrapper: React.FC<PodRingWrapperProps> = ({ workload }) => {
+  const workloadResource = React.useMemo(
+    () => ({
+      kind: workload?.kind,
+      namespace: workload?.namespace,
+      name: workload?.name,
+      optional: true,
+    }),
+    [workload],
+  );
+
+  const [workloadData] = useK8sWatchResource<K8sResourceKind>(workloadResource);
+
+  const podResource = React.useMemo(
+    () => ({
+      kind: PodModel.kind,
+      namespace: workload?.namespace,
+      selector: workloadData?.spec?.selector,
+      isList: true,
+      optional: true,
+    }),
+    [workload, workloadData],
+  );
```

Comment:
add checks for data is loaded or not
```suggestion
  const [workloadData, loaded, loadError] = useK8sWatchResource<K8sResourceKind>(workloadResource);

  const podResource = React.useMemo(
    () => (loaded && !loadError ? {
      kind: PodModel.kind,
      namespace: workload?.namespace,
      selector: workloadData?.spec?.selector,
      isList: true,
      optional: true,
    }: {}),
    [workload, workloadData, loaded, loadError],
  );
```

Reply:
cannot add empty object here